stdlib/universe/hour_selection.go

```diff
@@ -184,3 +208,29 @@ func (t *hourSelectionTransformation) UpdateProcessingTime(id execute.DatasetID,
 func (t *hourSelectionTransformation) Finish(id execute.DatasetID, err error) {
 	t.d.Finish(err)
 }
+
+func getLocation(args flux.Arguments) (string, values.Duration, error) {
+	var name, offset values.Value
+	var ok bool
+	if location, err := args.GetRequiredObject("location"); err != nil {
+		return "UTC", values.ConvertDurationNsecs(0), err
+	} else {
+		name, ok = location.Get("zone")
+		if !ok {
+			return "UTC", values.ConvertDurationNsecs(0), errors.New(codes.Invalid, "zone property missing from location record")
+		} else if got := name.Type().Nature(); got != semantic.String {
+			return "UTC", values.ConvertDurationNsecs(0), errors.Newf(codes.Invalid, "zone property for location must be of type %s, got %s", semantic.String, got)
+		}
+
+		if offset, ok = location.Get("offset"); ok {
+			if got := offset.Type().Nature(); got != semantic.Duration {
+				return "UTC", values.ConvertDurationNsecs(0), errors.Newf(codes.Invalid, "offset property for location must be of type %s, got %s", semantic.Duration, got)
+			}
+		}
+	}
+	if name.IsNull() {
+		name = values.NewString("UTC")
+	}
+
+	return name.Str(), offset.Duration(), nil
+}
```

Comment:
Is this copied and pasted from a function in the `date` package? Maybe that function could just be exported?